mlrun/utils/notifications/notification_pusher.py

```diff
@@ -236,13 +236,34 @@ def _prepare_notification_args(
         custom_message = (
             f": {notification_object.message}" if notification_object.message else ""
         )
-        message = self.messages.get(run.state(), "") + custom_message
+        resource = "Run"
+        runs = [run.to_dict()]
+
+        if "workflow" in run.metadata.labels:
+            resource = "Workflow"
+            custom_message = (
+                f" (workflow: {run.metadata.labels['workflow']}){custom_message}"
+            )
+            db = mlrun.get_run_db()
+
+            workflow_id = run.status.results.get("workflow_id", None)
+            if workflow_id:
+                workflow_runs = db.list_runs(
+                    project=run.metadata.project,
+                    labels=f"workflow={workflow_id}",
+                )
+                runs.extend(workflow_runs)
```

Comment:
why do you need all runs?

Reply:
I'm keeping the behaviour of how pipeline notifications always worked up until now and want to keep the content the same, only change the method of delivery.